kotlinx.coroutines.test.RunTestTest.testRunTestTimingOutAndThrowing[jvm] sporadically fails

https://teamcity.jetbrains.com/buildConfiguration/KotlinTools_KotlinxCoroutines_Build/4090249?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildTestsSection=true

```

java.lang.AssertionError: [] expected:<1> but was:<0>
  at org.junit.Assert.fail(Assert.java:89)
  at org.junit.Assert.failNotEquals(Assert.java:835)
  at org.junit.Assert.assertEquals(Assert.java:120)
  at kotlin.test.junit.JUnitAsserter.assertEquals(JUnitSupport.kt:32)
  at kotlin.test.AssertionsKt__AssertionsKt.assertEquals(Assertions.kt:63)
  at kotlin.test.AssertionsKt.assertEquals(Unknown Source)
  at kotlinx.coroutines.test.RunTestTest$testRunTestTimingOutAndThrowing$1.invoke(RunTestTest.kt:212)
  at kotlinx.coroutines.test.RunTestTest$testRunTestTimingOutAndThrowing$1.invoke(RunTestTest.kt:205)
  at kotlinx.coroutines.test.HelpersJvmKt.testResultMap(HelpersJvm.kt:7)
  at kotlinx.coroutines.test.RunTestTest.testRunTestTimingOutAndThrowing(RunTestTest.kt:205)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.base/java.lang.reflect.Method.invoke(Method.java:566)
```

--- a/kotlinx-coroutines-test/common/test/RunTestTest.kt
+++ b/kotlinx-coroutines-test/common/test/RunTestTest.kt
@@ -215,7 +215,7 @@ class RunTestTest {
             }
         }
     }) {
-        runTest(timeout = 1.milliseconds) {
+        runTest(timeout = 100.milliseconds) {
             coroutineContext[CoroutineExceptionHandler]!!.handleException(coroutineContext, TestException("A"))
             withContext(Dispatchers.Default) {
                 delay(10000)

